### PR TITLE
27-Jan

### PR DESCRIPTION
**Pull request by nileshjd2202**

### CODE REVIEW

File: listCompre.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+list1 = [i for i in range(start,ending+1)]
+
+dict1 = {x: x**2 for x in list1}
+
+alreadyNum = set()
+k = 0
+while(k<3):
+    user_input = int(input("Enter number to get square: "))
+    if user_input in dict1 :
```
Comment (MahavirAce) [inline listCompre.py]: take n time please reduce the complexity

File: listCompre.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        if user_input in alreadyNum:
+            print("Already entered this number")
+            k += 1
+        alreadyNum.add(user_input)
+        print("Your output: " + str(dict1[user_input]))
```
Comment (MahavirAce) [inline listCompre.py]: if number already visited why you print the output

File: listCompre.py (hunk 1 above)
Comment (nileshjd2202) [reply listCompre.py]: I have changed and used dict1.keys for constant time

File: listCompre.py (hunk 2 above)
Comment (nileshjd2202) [reply listCompre.py]: I have add else condition now it not print output again

File: listCompre.py [hunk 3]
```diff
@@ -10,13 +10,14 @@
 k = 0
 while(k<3):
     user_input = int(input("Enter number to get square: "))
-    if user_input in dict1 :
+    if user_input in dict1.keys :
```
Comment (MahavirAce) [inline listCompre.py]: is this tested?

**Pull request merged**